skstore/ts/src/prv/skstore_skjson.ts

```diff
@@ -13,34 +14,43 @@ export enum Type {
 }
 
 interface WasmAccess {
-  SKIP_SKJSON_typeOf: (json: ptr) => int;
-  SKIP_SKJSON_asInt: (json: ptr) => int;
-  SKIP_SKJSON_asFloat: (json: ptr) => float;
-  SKIP_SKJSON_asBoolean: (json: ptr) => boolean;
-  SKIP_SKJSON_asString: (json: ptr) => ptr;
-  SKIP_SKJSON_asArray: (json: ptr) => ptr;
-  SKIP_SKJSON_asObject: (json: ptr) => ptr;
-
-  SKIP_SKJSON_fieldAt: (json: ptr, idx: int) => ptr;
-  SKIP_SKJSON_get: (json: ptr, idx: int) => ptr;
-  SKIP_SKJSON_at: (json: ptr, idx: int) => ptr;
-
-  SKIP_SKJSON_objectSize: (json: ptr) => int;
-  SKIP_SKJSON_arraySize: (json: ptr) => int;
+  SKIP_SKJSON_typeOf: (json: ptr<Internal.CJSON>) => int;
+  SKIP_SKJSON_asInt: (json: ptr<Internal.CJSON>) => int;
+  SKIP_SKJSON_asFloat: (json: ptr<Internal.CJSON>) => float;
+  SKIP_SKJSON_asBoolean: (json: ptr<Internal.CJSON>) => boolean;
+  SKIP_SKJSON_asString: (json: ptr<Internal.CJSON>) => ptr<Internal.String>;
+  SKIP_SKJSON_asArray: (json: ptr<Internal.CJSON>) => ptr<Internal.CJArray>;
+  SKIP_SKJSON_asObject: (json: ptr<Internal.CJSON>) => ptr<Internal.CJObject>;
+
+  SKIP_SKJSON_fieldAt: (
+    json: ptr<Internal.CJObject>,
+    idx: int,
+  ) => ptr<Internal.String>; // Should be Opt<...>
```

Comment:
Are the semantics here that this comes back as a zero pointer if it's None on the skip side?

I ran into a few cases like that and added some "if the (skip) pointer is 0 just use a (javascript) null instead of trying to dereference" logic in previous PRs.  It's a bit odd how "Opt" in JS means "nullable" and not the usual tagged sum, but given that interpretation this seems fine as is -- do you agree or plan to make this an explicit option?

Reply:
I don't know, I've seen a few places where the `Option<T>` was translated into `Opt<ptr>` (btw `Opt<T>` is not standard Typescript, it is defined in `sk_types` as `T | null`).
Maybe @skiplabsdaniel knows more

Reply:
> btw `Opt<T>` is not standard Typescript, it is defined in `sk_types`

Yep, sorry, that's what I meant by "in JS".  I'm happy i you want to make it `Opt<ptr<...>>`, I just mean that the actual value there may be the number `0` not the value `null` -- so if we do make it `Opt`, the implementation should do that check on the value coming back from skip

Reply:
I'm not sure if it makes it to the js interface, but note that the skip option type is represented as a flat null pointer for `None` or the value of the `Some` case directly where possible. In C interop code one has to take account of that when manipulating values of option type. I don't see how the wasm code would be different, so that mapping to JS null is likely the most accurate.